Missionframework/scripts/shared/functions/F_kp_checkGear.sqf

```diff
@@ -19,44 +19,214 @@
-{_playerItems pushBackUnique _x;} forEach (assignedItems player);
-{_playerItems pushBackUnique _x;} forEach (uniformItems player);
-{_playerItems pushBackUnique _x;} forEach (vestItems player);
-{_playerItems pushBackUnique _x;} forEach (backpackItems player);
-{_playerItems pushBackUnique _x;} forEach (weapons player);
-{if (_x != "") then {_playerItems pushBackUnique _x;}} forEach (primaryWeaponItems player);
-{if (_x != "") then {_playerItems pushBackUnique _x;}} forEach (secondaryWeaponItems player);
-{if (_x != "") then {_playerItems pushBackUnique _x;}} forEach (handgunItems player);
-
-_playerItems = _playerItems apply {toLower _x};
-
-private _validItemsCount = {
-	(_x in KP_liberation_allowed_items)
-	|| ((_x find "acre") != -1)
-	|| ((_x find "tf_") != -1)
-	|| ((_x find "tfar_") != -1)
-} count _playerItems;
-
-if (_validItemsCount != count _playerItems) then {
-	private _text = format ["[KP LIBERATION] [BLACKLIST] Found %1 at Player %2", (_playerItems - KP_liberation_allowed_items), name player];
+private _removedItems = [];
+
+if (!(toLower(headgear player) in KP_liberation_allowed_items)) then {
+    _removedItems pushBack (headgear player);
+    removeHeadgear player;
+};
+if (!(toLower(goggles player) in KP_liberation_allowed_items)) then {
+    _removedItems pushBack (goggles player);
+    removeGoggles player;
+};
+if (!(toLower(uniform player) in KP_liberation_allowed_items)) then {
+    _removedItems pushBack (uniform player);
+    removeUniform player;
+};
+if (!(toLower(vest player) in KP_liberation_allowed_items)) then {
+    _removedItems pushBack (vest player);
+    removeVest player;
+};
+if (!(toLower(backpack player) in KP_liberation_allowed_items) && ((backpack player) != _backpack)) then {
+    _removedItems pushBack (backpack player);
+    removeBackpack player;
+};
+
+private _playerItems = assignedItems player;
+_playerItems append ((getItemCargo (uniformContainer player)) select 0);
+_playerItems append ((getItemCargo (vestContainer player)) select 0);
+_playerItems append ((getItemCargo (backpackContainer player)) select 0);
+_playerItems = _playerItems apply { toLower _x };
+{
+    _removedItems pushBack _x;
+    player unassignItem _x;
+    player removeItems _x;
+} forEach ((_playerItems arrayIntersect _playerItems) - KP_liberation_allowed_items);
+
+private _playerMagazines = ((getMagazineCargo (uniformContainer player)) select 0);
+_playerMagazines append ((getMagazineCargo (vestContainer player)) select 0);
+_playerMagazines append ((getMagazineCargo (backpackContainer player)) select 0);
+_playerMagazines = _playerMagazines apply { toLower _x };
+{
+    _removedItems pushBack _x;
+    player removeMagazines _x;
+} forEach ((_playerMagazines arrayIntersect _playerMagazines) - KP_liberation_allowed_items);
+
+// This is essentially slightly modified CBA_fnc_removeWeaponCargo because it's not possible to remove
+// weapons from containers (uniform, vest, backpack) with removeWeapon.
```

Comment:
Avoid inline functions.
Create a new function for this and call it.
Currently not reviewing the content of this function.

Is this also really needed? How often could it happen that players will try to "hide" an blacklisted weapon etc. in a backpack? (vest and uniform should be too small)
Personally I think in a PvP game mode this should be absolutely be checked. But in a coop game mode which is mostly played inside "closed groups", I don't know.

Reply:
it's mostly for sake of completeness I guess, although some people (me included) tend to stuff a bunch of single shot ATs in my backpacks from the enemy which could then be loaded in say a ww2 scenario and you'd have an unlimited supply of RPG-26's at your disposal... on the other hand you'd also have to put it in a backpack available in that scenario also...

Regardless, I moved the functions to their own files and tested that it works. In 0.97 forward I'd just use `CBA_fnc_removeWeaponCargo` instead but since we don't have a CBA dependency in this yet, I'm proposing this.